drivers: sensor: lm75: variable type & division

LorisSchmit opened the pull request:
Using an int16_t for the temp variable and integer division
for the temperature conversion results in a max resolution of +/- 0.5°C.
This is fine for 9-bit LM75 sensor, however for 10, 11 and 12-bit sensor
(like the Atmel AT30TS75A) a float as type for the temp variable and
float division for the temperature conversion would be more suitable.
This allows to display the temperature to the up to +/- 0.0625 °C accuracy.

Signed-off-by: Loris Schmit <loris.schmit@inovex.de>

Commit:
drivers: sensor: lm75: variable type & division

Using an int16_t for the temp variable and integer division
for the temperature conversion results in a max resolution of +/- 0.5°C.
This is fine for 9-bit LM75 sensor, however for 10, 11 and 12-bit sensor
(like the Atmel AT30TS75A) a float as type for the temp variable and
float division for the temperature conversion would be more suitable.
This allows to display the temperature to the up to +/- 0.0625 °C accuracy.

Signed-off-by: Loris Schmit <loris.schmit@inovex.de>

henrikbrixandersen reviewed (changes requested):
We usually do not use floating point math in the sensor drivers themselves. Can this be done using fixed-point?

Commit:
drivers: sensor: lm75: fixed point var & temp conv

temp variable is fixed-point now.
Also the temperature conversion is now generalized
for all resolutions (9,10,11,12 bit).
A device property was added to configure the temperature
resolution. This is only possible for the Atmel AT30TS75A
right now, as most LM75-standard temperature sensor don't
feature a user-configurable temperature resolution.

Signed-off-by: Loris Schmit <loris.schmit@inovex.de>
